### PR TITLE
User filter

### PR DESCRIPTION
**Pull request by vinhtruong94**
Added and modified addressVerified for UserFilter, added some fields …
…for validation in IndexUserRequest, added scope method scopeWithRole for user role in User model, added getUsers for userResource.svc.js to get users from different filters, addedhtml buttons and html select options for user Filter in html file,added search functions for filters for userSearchBar js based on motionSearchBar ( motion Filter)

### CODE REVIEW

File: app/Filters/UserFilter.php [hunk 1]
```diff
@@ -38,6 +38,11 @@ public function allNames($name = '')
                            ->orWhere('last_name', 'like', "%$name%");
     }
 
+    public function role($role = '')
```
Comment (psaunders) [inline app/Filters/UserFilter.php]: There is already a "roles" query below, you need to pass an array like motion statuses

File: app/Http/Requests/User/IndexUserRequest.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ public function rules()
             'allNames'                => 'string',
             'page'                    => 'integer',
             'limit'                   => 'integer',
-            'roles'                   => 'array',
```
Comment (psaunders) [inline app/Http/Requests/User/IndexUserRequest.php]: It should take roles, people can have 1:m roles

File: app/User.php [hunk 3]
```diff
@@ -524,6 +524,13 @@ public function scopeDoesntHaveRoles($query, $role)
         });
     }
 
+    public function scopeWithRoles($query, $role)
```
Comment (psaunders) [inline app/User.php]: It's just below it, why all the duplication. Just query with an array

File: public/app/components/userSidebar/search/userSearchbar.tpl.html [hunk 4]
```diff
@@ -12,11 +12,39 @@
...
 
   <div ng-class="search.isOpen?'':'action-buttons'" layout-align="end end">
     <md-icon class="mdi mdi-magnify search-icon" ng-click="search.show()" ng-class="{'search-icon':search.isOpen}"></md-icon>
 
+      <md-menu md-offset="27 32">
+        <md-icon class="mdi mdi-dots-vertical" ng-click="$mdOpenMenu()"></md-icon>
+        <md-menu-content width="4" class="md-body-1">
+
+            <md-select placeholder="Filter by Role" ng-model="search._role.filter" ng-change="search.searchSpecific()" aria-label="filter users by roles" layout-padding>
+              <md-option ng-value="search._role.filters[0].query">{{search._role.filters[0].name}}</md-option>
```
Comment (psaunders) [inline public/app/components/userSidebar/search/userSearchbar.tpl.html]: Should be a loop so changing it one place changes it here

File: public/app/components/userSidebar/search/userSearchbar.tpl.html [hunk 5]
```diff
@@ -12,11 +12,39 @@
...
+
+              
+            </md-select>  
+            <md-select placeholder="Filter by Identity" ng-model="search._identity.filter" ng-change="search.searchSpecific()" aria-label="filter users by identity" layout-padding>
+              <md-option ng-value="search._identity.filters[0].query">{{search._identity.filters[0].name}}</md-option>
+              <md-option ng-value="search._identity.filters[1].query">{{search._identity.filters[1].name}}</md-option>
+            </md-select> 
+            
+            <md-select placeholder="Filter by Address" ng-model="search._addressVerified.filter" ng-change="search.searchSpecific()" aria-label="filter users by address" layout-padding>
+              <md-option ng-value="search._addressVerified.filters[0].query">{{search._addressVerified.filters[0].name}}</md-option>
```
Comment (psaunders) [inline public/app/components/userSidebar/search/userSearchbar.tpl.html]: What's with the underscores?

**Comment by vinhtruong94** [comment]
Undo redundant back-end codes. 
Added back roles validation in IndexUserRequest.
Tight up html to loop repeat instead of multi underscores.
Added front end js query for roles.
Tested on local front end, all features and buttons are functioned.
Please review the changes, you can pull to review new functions.

**Pull request merged**